src/modules/job-list/idsync.c

```diff
@@ -54,6 +54,7 @@ static struct idsync_data *idsync_data_create (flux_t *h,
                                                flux_jobid_t id,
                                                const flux_msg_t *msg,
                                                json_t *attrs,
+                                               flux_job_state_t state,
```

Comment:
Commit message: it might be fine, but I was stumbling over reading of:
> Solution: Support an option job state in the job-list.list-id service.

Is it supposed to be either "Support an optional job state in the..." or maybe "Support a job state option in the..."?